specs/governance/gov-token.md

```diff
@@ -24,72 +24,82 @@
 
 ### Token Burning
 
 The contract MUST allow token holders to burn their own tokens using the inherited `burn(uint256)` or
 `burnFrom(address,uint256)` functions inherited from `ERC20Burnable`. When tokens are burned, the total supply and the
-holder's voting power MUST be reduced accordingly.
+holder's voting power MUST be reduced accordingly in the `Alligator` contract via the transfer hook.
 
 ### Voting Power
 
 Each token corresponds to one unit of voting power.
 By default, token balance does not account for voting power. To have their voting power counted, token holders MUST delegate
 their voting power to an address (can be their own address).
-The contract MUST offer public accessors for quering voting power, as outlined below.
+The contract MUST offer public accessors for querying voting power, as outlined below.
 
 #### Queries
 
-- The `getVotes()(uint256)` function MUST return the current voting power of an address.
+- The `getVotes()(uint256)` function MUST retrieve the current voting power of an address from the `Alligator` contract.
 - The `getPastVotes(address,uint256)(uint256)` function MUST allow querying the voting power of an address at a specific
-  block number in the past.
+  block number in the past from the `Alligator` contract.
 - The `getPastTotalSupply(uint256)(uint256)` function MUST return the total voting power at a specific block number in
-  the past.
+  the past from the `Alligator` contract.
 
 ### Delegation
 
 Vote power can be delegated either by calling the `delegate(address)` function directly (to delegate as the `msg.sender`)
 or by providing a signature to be used with function `delegateBySig(address,uint256,uint256,uint8,bytes32,bytes32)`,
-as inherited from `ERC20Votes`.
+as inherited from `ERC20Votes`. These functions are modified to forward the calls to the `Alligator` contract using an
+advanced delegation rule that mimics basic delegation.
+
+Advanced delegation features, such as partial delegation and subdelegations, are handled by the `Alligator` contract.
+Partial delegation allows delegators to distribute their voting power among multiple delegatees in a fractional manner,
+while subdelegations enable delegatees to further delegate their received voting power to other delegatees based on predefined
+rules and constraints.
 
-The delegation is recorded in a checkpoint. When a token holder delegates their voting power, the delegated address receives
-the voting power corresponding to the token holder's balance. Under token transfers, the voting power associated with the
-tokens is moved from the delegate of the sender to that of the recipient.
+The `Alligator` contract maintains the necessary invariants, such as preventing circular delegation chains, ensuring vote
+weight consistency, and managing checkpoints. It is incorporated as a predeploy of the OP stack to avoid manual deployments
+across the Superchain.
```

Comment:
Is there going to be another spec for `Alligator`? There should be, since it will become a predeploy. Related, the design doc had a lot of invariants that we should ensure are captured in the spec somewhere.

Reply:
yes, there's going to be another spec for `Alligator`, which are tracked by these issues: https://github.com/ethereum-optimism/protocol-quest/issues/297 & https://github.com/ethereum-optimism/protocol-quest/issues/298

Reply:
Perfect, thanks! Will resolve this, since the invariants were all properties of Alligator